src/core/processing/qgsprocessingparametermeshdataset.h

```diff
@@ -0,0 +1,215 @@
+/***************************************************************************
+  qgsprocessingparametermeshdataset.h
+  ---------------------
+  Date                 : October 2020
+  Copyright            : (C) 2020 by Vincent Cloarec
+  Email                : vcloarec at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSPROCESSINGPARAMETERMESHDATASET_H
+#define QGSPROCESSINGPARAMETERMESHDATASET_H
+
+#include "qgsprocessingparameters.h"
+#include "qgsprocessingparametertype.h"
+#include "qgsmeshdataset.h"
+
+/**
+ * A parameter for processing algorithms that need a list of mesh dataset groups
+ * A valid value for this parameter is a list (QVariantList) of dataset groups index in the mesh layer scope
+ *
```

Comment:
You should describe here how you expect algorithms to evaluate the parameter value. Ideally add a "valueAsDatasetGroupIndexes" method to this class too, to make that easier for callers. Same applies to QgsProcessingParameterTypeMeshDatasetGroups

Reply:
I think this is fixed by valueAsDatasetGroup